presidio-image-redactor/presidio_image_redactor/qr_recognizer.py

```diff
@@ -0,0 +1,144 @@
+import cv2
+import numpy as np
+
+
+class QRRecognizerResult:
+    """
+    Represent the results of analysing the image by QRRecognizer.
+
+    :param text: Decoded text
+    :param bbox: Bounding box in the following format - [left, top, width, height]
+    :param polygon: Polygon aroung QR code
+    """
+
+    def __init__(
+        self,
+        text: str,
+        bbox: Tuple[int, int, int, int],
+        polygon: Optional[List[int]] = None,
+    ):
+        self.text = text
+        self.bbox = bbox
+        self.polygon = polygon
+
+    def __eq__(self, other):
+        """
+        Compare two QRRecognizerResult objects.
+
+        :param other: another QRRecognizerResult object
+        :return: bool
+        """
+        equal_text = self.text == other.text
+        equal_bbox = self.bbox == other.bbox
+        equal_polygon = self.polygon == other.polygon
+
+        return equal_text and equal_bbox and equal_polygon
+
+    def __repr__(self) -> str:
+        """Return a string representation of the instance."""
+        return (
+            f"{type(self).__name__}("
+            f"text={self.text}, "
+            f"bbox={self.bbox}, "
+            f"polygon={self.polygon})"
+        )
+
+
+class QRRecognizer(ABC):
+    """
+    A class representing an abstract QR code recognizer.
+
+    QRRecognizer is an abstract class to be inherited by
+    recognizers which hold the logic for recognizing QR codes on the images.
+    """
+
+    @abstractmethod
+    def recognize(self, image: object) -> List[QRRecognizerResult]:
+        """Detect and decode QR codes on the image.
+
+        :param image: PIL Image/numpy array to be processed
+
+        :return: List of the recognized QR codes
+        """
+
+
+class OpenCVQRRecongnizer(QRRecognizer):
+    """
+    QR code recognition using OpenCV.
+
+    Example of the usage:
+        from presidio_image_redactor import OpenCVQRRecognizer
+
+        image = cv2.imread("qrcode.jpg")
+        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+
+        recognized = OpenCVQRRecongnizer().recognize(image)
+    """
+
+    def __init__(self) -> None:
+        self.detector = cv2.QRCodeDetector()
+
+    def recognize(self, image: object) -> List[QRRecognizerResult]:
+        """Detect and decode QR codes on the image.
+
+        :param image: PIL Image/numpy array to be processed
+
+        :return: List of the recognized QR codes
+        """
+
+        if not isinstance(image, np.ndarray):
+            image = np.array(image, dtype=np.uint8)
+
+        recognized = []
+
+        ret, points = self._detect(image)
+
+        if ret:
+            decoded = self._decode(image, points)
+
+            for text, p in zip(decoded, points):
+                (x, y, w, h) = cv2.boundingRect(p)
+
+                recognized.append(
+                    QRRecognizerResult(
+                        text=text, bbox=[x, y, w, h], polygon=[*p.flatten(), *p[0]]
+                    )
+                )
```

Comment:
Prefer immutability.
```suggestion
            for text, p in zip(decoded, points):
                (x, y, w, h) = cv2.boundingRect(p)

                recognized.append(
                    QRRecognizerResult(
                        text=text, bbox=[x, y, w, h], polygon=[*p.flatten(), *p[0]]
                    )
                )
```
recognized = [QRRecognizerResult(text=text, bbox=cv2.boundingRect(point), polygon=[*point.flatten(), *point[0]]) for text, point in zip(decoded, points)]

Reply:
I will add these changes, thanks for the suggestion.